src/fn_call.rs

```diff
@@ -831,3 +1039,83 @@ fn gen_random<'a, 'mir, 'tcx>(
         }
     }
 }
+
+/// A helper method to unwind the stack.
+///
+/// We execute the 'unwind' blocks associated with frame
+/// terminators as we go along (these blocks are responsible
+/// for dropping frame locals in the event of a panic)
+///
+/// When we find our target frame, we write the panic payload
+/// directly into its locals, and jump to it.
+/// After that, panic handling is done - from the perspective
+/// of the caller of '__rust_maybe_catch_panic', the function
+/// has 'returned' normally, after which point Miri excecution
+/// can proceeed normally.
+fn unwind_stack<'a, 'mir, 'tcx>(
+    this: &mut MiriEvalContext<'a, 'mir, 'tcx>,
+    payload_data_ptr: Scalar<Borrow>,
+    payload_vtable_ptr: Scalar<Borrow>
+) -> EvalResult<'tcx> {
+    while !this.stack().is_empty() {
```

Comment:
I am a bit surprised, isn't there code this has to execute? Like, all the stack frames we are popping have stuff they need to `drop`, where is this doing that?

Reply:
That's handled [here](https://github.com/rust-lang/miri/pull/693/files/1d263246eb67a806ca4e61f2b13705e0d5eae81f#diff-6bac323bc5f550c3b0997b72a303e450R1091)

Reply:
That link doesn't seem to work any more... links into reviews are quite bad on GH, could you link into a file in a commit ([like this one](https://github.com/rust-lang/miri/blob/4893036a34d9f979a660290151c19c37efae9002/src/fn_call.rs)) instead?